subsys/net/l2/ppp/ipcp.c

```diff
@@ -147,6 +147,33 @@ struct ipcp_peer_option_data {
 	struct in_addr addr;
 };
 
+#if defined(CONFIG_PPP_DIALUP)
+static int ipcp_dns_address_parse(struct ppp_fsm *fsm, struct net_pkt *pkt,
+				 void *user_data)
+{
+	struct ipcp_peer_option_data *data = user_data;
+	struct sockaddr_in zeroes;
+	int ret;
+
+	ret = net_pkt_read(pkt, &data->addr, sizeof(data->addr));
+	if (ret < 0) {
+		/* Should not happen, is the pkt corrupt? */
+		return -EMSGSIZE;
+	}
+
+	/* Request is zeros? Give our dns address in ConfNak */
+	if (net_addr_pton(AF_INET, "0.0.0.0", &zeroes) >= 0 &&
```

Comment:
This will not work, the `dst` option in `net_addr_pton()` is of type `struct in_addr` and not `struct sockaddr_in`.
You can write this simpler way like this:
```
if (data->addr == INADDR_ANY) {
```

Reply:
thanks, a good catch. it was actually working, now wondering why.